core/phase.go

```diff
@@ -293,6 +301,21 @@ func checkPhaseCompletion(
 			return
 		}
 
+		duration := train.DeployedAt.Value.Sub(train.CreatedAt.Value)
+
+		options, err := dataClient.Options()
+		if err != nil {
```

Comment:
Is this for recording metrics?  If so, exiting on an error related to metric gathering would fail the rest of the actual work.  Logging the error, skipping the metric generation and continuing would be safer.